aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -526,6 +529,188 @@ void cpu_hflip_vec(at::TensorIterator& iter) {
 }
 
 
+template<int size0>
```

Comment:
Not sure about this being a template parameter, how much does it effect performance?

Reply:
Yes, it effects at some point. Without template param, channels=3 is a bit slower than Pillow and with template it is a bit faster than Pillow. In numbers on my machine it is the following : 
```
Pillow : 48 us
PR with template : 41 us
PR without template : 54 us
```

Mask generation is not expensive, IMO.